cmd/pbackend/statuser.go

```diff
@@ -175,13 +176,15 @@ func checkSourceAvailabilityAWS(cancelCtx context.Context) {
 			if err != nil {
 				sr.Status = kafka.StatusUnavailable
 				sr.Err = err
+				sr.UserError = "Could not log into AWS account"
 				logger.Warn().Err(err).Msg("Could not get aws assumed client")
 			} else {
 				sr.Status = kafka.StatusAvailable
 				permissions, err = ec2Client.CheckPermission(ctx, &s.Authentication)
 				if err != nil {
 					sr.Status = kafka.StatusUnavailable
 					sr.Err = err
+					sr.UserError = fmt.Sprintf("Missing AWS permissions %v", permissions)
```

Comment:
Sure, How do you want to decide on a length?

Reply:
I'd pick a random number and then tested it in the UI how it handles it? :thinking:
I'd go with something like 150 characters.

Ultimatelly tho I believe it might be better to be handled on Sources side, don't they have some truncation for the UI? Wouldn't that be enough?

Reply:
This would end up with an output like: `Missing AWS permissions [perm1 perm2 perm3]`, right? We probably want only "Missing AWS permissions: perm1, perm2, perm3".
I have done this, too, and Lukas made a valid point about us wanting this displayed as a string. So I suggest that we can either make a function, something like a "PermissionsToString". An int parameter like "len" could solve even what Ondrej suggests. 
`func (c *ec2Client) PermissionsToString(ctx context.Context, []string permissions, int len) (string, error) {`
And I need some shortening and string conversion even for the UI change. What do we think? :thinking:

Reply:
I think it will be better practise to create a new struct named Permissions which is an array of strings, and have a receiver function that converts is to s string as you described. 

```
type Permissions []string

func (p Permissions) String() string {
	return "...."
}
```
But this means we would need to refactor some of the code, every []string would change to Permissions struct. 
What do you think?